src/middlewares/Mock.spec.ts

```diff
@@ -0,0 +1,128 @@
+import {Client, IResponse} from "../Client";
+import {Mock} from "./Mock";
+describe("Mock", () => {
+    it("is defined", () => {
+        expect(Mock).toBeDefined();
+    });
+
+    describe("constructor()", () => {
+        it("accepts IHandler array in constructor", () => {
+            const obj = new Mock([
+                {
+                    delay: 0,
+                    match: undefined as any,
+                    factory: undefined as any
+                }
+            ]);
+            expect(obj).toBeDefined();
+        });
+    });
+
+    describe("addHandler", () => {
+        it("is a a function", () => {
+            expect(typeof (new Mock()).addHandler).toBe("function");
+        });
+
+        it("is able to add handler to array", () => {
+            const obj = new Mock();
+            obj.addHandler({match: undefined as any, delay: 5, factory: undefined as any});
```

Comment:
Hmm that's a valid problem.

How about the following:
1) rename addHandler to registerHandler, check if the reference is already in the list when adding. return true on success.
2) add a unregisterHandler(handler) to remove a handler. return true on success
3) add resetHandlers() to remove all registered handlers.
 
Maybe this is a bit too much, but maybe its an interesting scenario.

Otherwise for testing the add function: 
1) add something with a useful function
2) invoke process() 
3) see if your handler actually got called.